agent/pool/pool.go

```diff
@@ -387,6 +389,26 @@ func (p *ConnPool) dial(
 	return conn, hc, nil
 }
 
+func (p *ConnPool) RPCClientTimeout() time.Duration {
+	// Since clientTimeout represents uint32 milliseconds,
+	// convert to int64 first then multiply by 1000 to
+	// match time.Duration's nanoseconds.
+	return time.Duration(int64(atomic.LoadUint32(p.clientTimeout) * 1000))
+}
+
+func (p *ConnPool) SetRPCClientTimeout(timeout time.Duration) {
+	if timeout/1000 > time.Duration(^uint32(0)) {
```

Comment:
I don't think this is right - `time.Duration` is expressed in nanoseconds and there are 1 million in a millisecond not 1000 so this seems off by three orders of magnitude.

In general you should never need to use hard-coded numbers like `1000` when working with durations in Go - the built in `time.Millisecond` etc. are more readable.

I think overall rather than trying to work out the absolute largest timeout we can accept which is ludicrously large I'd be tempted to make this whole method way simpler with a reasonable upper bound, something like:

```go

func (p *ConnPool) SetRPCClientTimeout(timeout time.Duration) {
  if timeout > time.Hour {
    // Prevent unreasonably large timeouts that might overflow a uint32
    timeout = time.Hour
  }
  
  atomic.StoreUint32(&p.clientTimeout, uint32(timeout.Milliseconds()))
}
```

Reply:
That sounds more reasonable and readable!